kitsune/users/views.py

```diff
@@ -246,16 +247,20 @@ def edit_watch_list(request):
 def edit_profile(request, username=None):
     """Edit user profile."""
     # If a username is specified, we are editing somebody else's profile.
-    if username and username != request.user.username:
-        try:
-            user = User.objects.get(username=username)
-        except User.DoesNotExist:
-            raise Http404
+    user = None
+    if username:
+        username = username.replace(" ", "+")
+
+        if username != request.user.username:
+            try:
+                user = User.objects.get(username=username)
+            except User.DoesNotExist:
+                raise Http404
 
-        # Make sure the auth'd user has permission:
-        if not request.user.has_perm("users.change_profile"):
-            return HttpResponseForbidden()
-    else:
+            # Make sure the auth'd user has permission:
+            if not request.user.has_perm("users.change_profile"):
+                return HttpResponseForbidden()
+    if not user:
```

Comment:
I think this can remain an `else`: the only case in which the above block leaves `user` falsy it will have raised `Http404`. Then L250 can also be removed.

Reply:
The problem here is that the username is passed and the `else` clause was activated because the second comparison of the `and` operator was False. With the new change it will get into the username but not in the second if thus it will never reach the `else` clause producing an unbound variable error from line 267

Reply:
Oh of course, I've oversimplified, ignore me!